src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.CSharpCodeGenerator.cs

```diff
@@ -667,42 +702,39 @@ protected StatementSyntax GetStatementContainingInvocationToExtractedMethodWorke
                 }
 
                 var syntaxNode = originalDocument.Root.GetAnnotatedNodesAndTokens(MethodDefinitionAnnotation).FirstOrDefault().AsNode();
-                if (syntaxNode == null || !(syntaxNode is MethodDeclarationSyntax methodDeclaration))
+                if (syntaxNode == null || !(syntaxNode is MethodDeclarationSyntax || syntaxNode is LocalFunctionStatementSyntax))
                 {
                     return await base.UpdateMethodAfterGenerationAsync(originalDocument, methodSymbolResult, cancellationToken).ConfigureAwait(false);
                 }
 
                 var semanticModel = originalDocument.SemanticModel;
-                var methodOperation = semanticModel.GetOperation(methodDeclaration, cancellationToken);
 
-                var returnOperations = methodOperation.DescendantsAndSelf().OfType<IReturnOperation>();
-
-                foreach (var returnOperation in returnOperations)
+                if (syntaxNode is MethodDeclarationSyntax methodDeclaration)
```

Comment:
it's really unfortunate to have all this special casing for MethodDeclaration and LocalFunctionDeclaration.  A better way to do this would be to make this class abstract and generic (parameterized on something like TMethodDeclaration).  Then have the core algorithms work uniformly on this, and extract MethodDecl/LocalFunc specific behavior to the subclasses.